pkg/server/plugin/nodeattestor/awsiid/organization.go

```diff
@@ -0,0 +1,251 @@
+	orgListAccountMap             map[string]any
+	orgListAccountMapCreationTime time.Time
+	orgConfig                     *orgValidationConfig
+	mutex                         sync.RWMutex
+	// orgAccountListCacheTTL holds the cache ttl from configuration; otherwise, it will be set to the default value.
+	orgAccountListCacheTTL time.Duration
+	log                    hclog.Logger
+	// retries fix number of retries before ttl is expired.
+	retries int
+}
+
+func newOrganizationValidationBase(config *orgValidationConfig) *orgValidator {
+	client := &orgValidator{
+		orgListAccountMap: make(map[string]any),
+		orgConfig:         config,
+		retries:           orgAccountRetries,
+	}
+
+	return client
+}
+
+func (o *orgValidator) getRetries() int {
+	o.mutex.RLock()
+	defer o.mutex.RUnlock()
+	return o.retries
+}
+
+func (o *orgValidator) decrRetries() int {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+	if o.retries > 0 {
+		o.retries--
+	}
+
+	return o.retries
+}
+
+func (o *orgValidator) configure(config *orgValidationConfig) error {
+	o.mutex.Lock()
+	defer o.mutex.Unlock()
+
+	o.orgConfig = config
+	// required for the way existing getClient function is designed
+	o.orgConfig.AccountRegion = "us-west-2"
+
+	// While doing configuration invalidate the map so we dont keep using old one.
+	o.orgListAccountMap = make(map[string]any)
+	o.retries = orgAccountRetries
+
+	t, err := time.ParseDuration(config.AccountListTTL)
+	if err != nil {
+		return status.Errorf(codes.InvalidArgument, "issue while parsing ttl for organization, while configuring orgnization validation: %v", err)
+	}
+
+	o.orgAccountListCacheTTL = t
+
+	return nil
+}
+
+func (o *orgValidator) setLogger(log hclog.Logger) {
+	o.log = log
+}
+
+// IsMemberAccount method checks if the Account ID attached on the node is part of the organisation.
+// If it part of the organisation then validation should be succesfull if not attestation should fail, on enabling this verification method.
+// This could be alternative for not explicitly maintaining allowed list of account ids.
+// Method pulls the list of accounts from organization and caches it for certain time, cache time can be configured.
+func (o *orgValidator) IsMemberAccount(ctx context.Context, orgClient organizations.ListAccountsAPIClient, accoundIDOfNode string) (bool, error) {
+	reValidatedcache, err := o.validateCache(ctx, orgClient)
+	if err != nil {
+		return false, err
+	}
+
+	accountIsmemberOfOrg, err := o.lookupCache(ctx, orgClient, accoundIDOfNode, reValidatedcache)
+	if err != nil {
+		return false, err
+	}
+
+	return accountIsmemberOfOrg, nil
+}
+
+// validateCache validates cache and refresh if its stale
+func (o *orgValidator) validateCache(ctx context.Context, orgClient organizations.ListAccountsAPIClient) (bool, error) {
+	isStale := o.checkIfOrgAccountListIsStale(ctx)
+	// refresh the account map
+	if isStale {
+		_, err := o.reloadAccountList(ctx, orgClient, false)
+		if err != nil {
+			return true, err
```

Comment:
it si only going to return true when error, is that right?
if that is the case, what is the value of boolean?
may we just return error?

if not we need another return with true when there is no error, something like:
```
        isStale := o.checkIfOrgAccountListIsStale(ctx)
        if !isStale {
        	return false, nil
        }

        // Cache is stale, reload it
        _, err := o.reloadAccountList(ctx, orgClient, false)
	if err != nil {
		return false, err
        }
        
        return true, nil
```